### PR TITLE
document virtualColumns in native query documentation, fix some redirects

### PR DESCRIPTION
**Pull request by clintropolis**
### Description
While putting together a PR to document the `virtualColumns` support added to native search queries in #12720, i noticed that none of the native queries had this documented, so I added it linking to https://github.com/apache/druid/blob/master/docs/querying/virtual-columns.md.

I ran the build locally which provided some warnings about redirects which I also fixed while I was here
```
On line 35 could not find anchor 'firehoses-deprecated' in 'ingestion/native-batch.html'
On line 160 source ingestion/compaction.html already exists
On line 160 could not find anchor 'compact' in 'ingestion/data-management.html'
On line 162 could not find anchor 'firehoses-deprecated' in 'ingestion/native-batch.html'
On line 163 could not find anchor 'flattenspec' in 'ingestion/index.html'
On line 165 source ingestion/ingestion-spec.html already exists
On line 165 could not find anchor 'spec' in 'ingestion/index.html'
On line 174 could not find anchor 'schema-changes' in 'ingestion/data-management.html'
On line 178 could not find anchor 'transformspec' in 'ingestion/index.html'
There are 9 issues with the redirects
```

Running the build regenerated the i18n/en.json file, which i just left in here, but I can revert if anyone cares since I don't think it matters that much because _i think_ it would also be generated when we do an actual website build to publish stuff.

### CODE REVIEW

File: docs/querying/groupbyquery.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ Following are main parts to a groupBy query:
 |queryType|This String should always be "groupBy"; this is the first thing Druid looks at to figure out how to interpret the query|yes|
 |dataSource|A String or Object defining the data source to query, very similar to a table in a relational database. See [DataSource](../querying/datasource.md) for more information.|yes|
 |dimensions|A JSON list of dimensions to do the groupBy over; or see [DimensionSpec](../querying/dimensionspecs.md) for ways to extract dimensions. |yes|
+|virtualColumns|Define any [virtual columns](./virtual-columns.md) to be available to use as `dimensions` or referenced as inputs to `aggregations` or `postAggregations`.| no (default none)|
```
Comment (vtlim) [inline docs/querying/groupbyquery.md:89]:
```suggestion
|virtualColumns|A JSON list of [virtual columns](./virtual-columns.md). You can reference the virtual columns in `dimensions`, `aggregations`, or `postAggregations`.| no (default none)|
```

File: docs/querying/searchquery.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ There are several main parts to a search query:
 |limit| Defines the maximum number per Historical process (parsed as int) of search results to return. |no (default to 1000)|
 |intervals|A JSON Object representing ISO-8601 Intervals. This defines the time ranges to run the query over.|yes|
 |searchDimensions|The dimensions to run the search over. Excluding this means the search is run over all dimensions.|no|
+|virtualColumns|Define any [virtual columns](./virtual-columns.md) to be available to use as `searchDimensions`.| no (default none)|
```
Comment (vtlim) [inline docs/querying/searchquery.md:65]:
```suggestion
|virtualColumns|A JSON list of [virtual columns](./virtual-columns.md) available to use in `searchDimensions`.| no (default none)|
```

File: docs/querying/timeseriesquery.md [hunk 3]
```diff
@@ -78,6 +78,7 @@ There are 7 main parts to a timeseries query:
 |intervals|A JSON Object representing ISO-8601 Intervals. This defines the time ranges to run the query over.|yes|
 |granularity|Defines the granularity to bucket query results. See [Granularities](../querying/granularities.md)|yes|
 |filter|See [Filters](../querying/filters.md)|no|
+|virtualColumns|Define any [virtual columns](./virtual-columns.md) to be available to referenced as inputs to `aggregations` or `postAggregations`.| no (default none)|
```
Comment (vtlim) [inline docs/querying/timeseriesquery.md:81]:
```suggestion
|virtualColumns|A JSON list of [virtual columns](./virtual-columns.md). You can reference the virtual columns in `aggregations` or `postAggregations`.| no (default none)|
```

File: docs/querying/topnquery.md [hunk 4]
```diff
@@ -103,6 +103,7 @@ There are 11 parts to a topN query.
 |intervals|A JSON Object representing ISO-8601 Intervals. This defines the time ranges to run the query over.|yes|
 |granularity|Defines the granularity to bucket query results. See [Granularities](../querying/granularities.md)|yes|
 |filter|See [Filters](../querying/filters.md)|no|
+|virtualColumns|Define any [virtual columns](./virtual-columns.md) to be available to use as the grouping `dimension` or referenced as inputs to `aggregations` or `postAggregations`.| no (default none)|
```
Comment (vtlim) [inline docs/querying/topnquery.md:106]:
```suggestion
|virtualColumns|A JSON list of [virtual columns](./virtual-columns.md). You can reference a virtual column as the grouping `dimension` or as an input in `aggregations` or `postAggregations`.| no (default none)|
```

File: website/redirects.json [hunk 5]
```diff
@@ -171,11 +169,11 @@
 {"source": "ingestion/overview.html", "target": "index.html"}
 {"source": "ingestion/realtime-ingestion.html", "target": "index.html"}
 {"source": "ingestion/reports.html", "target": "tasks.html#reports"}
-{"source": "ingestion/schema-changes.html", "target": "data-management.html#schema-changes"}
+{"source": "ingestion/schema-changes.html", "target": "faq.html#how-can-i-reindex-existing-data-in-druid-with-schema-changes"}
```
Comment (vtlim) [inline website/redirects.json:172]: I think the target should be `design/segments.html#segments-with-different-schemas`.

The source of `schema-changes.html` was probably some version of https://apache.googlesource.com/druid/+/refs/heads/0.13.0-incubating/docs/content/ingestion/schema-changes.md.

The same content is in https://github.com/apache/druid/blob/master/docs/design/segments.md

Comment (clintropolis) [reply website/redirects.json]: sgtm, changed

**Review by vtlim — APPROVED** [review]
LGTM 🦖

**Pull request merged**